storage/netapp_e/na_eseries_storage_system.py

```diff
@@ -0,0 +1,319 @@
+    meta_tags = p['meta_tags']
+    enable_trace = p['enable_trace']
+
+    try:
+        api_usr = p['api_username']
+        api_pwd = p['api_password']
+        api_url = p['api_url']
+    except KeyError:
+        api_usr = os.environ['API_USERNAME']
+        api_pwd = os.environ['API_PASSWORD']
+        api_url = os.environ['API_URL']
+
+    if not api_usr or not api_pwd or not api_url:
+        module.fail_json(msg="You must pass in api_username "
+                             "and api_password and api_url to the module. Alternatively "
+                             "you can specify API_URL, API_USERNAME and API_PASSWORD "
+                             "as environment variables.")
+
+    changed = False
+    array_exists = False
+
+    try:
+        (rc, resp) = request(api_url + "/storage-systems/{0}".format(ssid), headers=dict(Accept="application/json"),
+                             url_username=api_usr, url_password=api_pwd, validate_certs=validate_certs,
+                             ignore_errors=True)
+    except:
+        err = get_exception()
+        module.fail_json(msg="Error accessing storage-system with id [{0}]. Error [{1}]".format(ssid, str(err)))
+
+    if rc == 200:
+        array_exists = True
+        array_detail = resp
+
+        if state == 'absent':
+            changed = True
+            array_exists = False
+        elif state == 'present':
+            current_addresses = set(
+                [i for i in [array_detail['ip1'], array_detail['ip2']] if i is not None and len(i) > 1])
+            if set(controller_addresses) != current_addresses:
+                changed = True
+            if array_detail['wwn'] != array_wwn and array_wwn != None:
+                module.fail_json(
+                    msg='It seems you may have specified a bad WWN. The storage system ID you specified, %s, currently has the WWN of %s' % (
+                        ssid, array_detail['wwn']))
+    elif rc == 404:
+        if state == 'present':
+            changed = True
+            array_exists = False
+        else:
+            changed = False
+            module.exit_json(changed=changed, msg="Storage system was not present.")
+
+    if changed and not module.check_mode:
+        if state == 'present':
+            if not array_exists:
+                # add the array
+                array_add_req = dict(
+                    id=ssid,
+                    controllerAddresses=controller_addresses,
+                    metaTags=meta_tags,
+                    enableTrace=enable_trace
+                )
+
+                if array_wwn:
+                    array_add_req['wwn'] = array_wwn
+
+                if array_password:
+                    array_add_req['password'] = array_password
+
+                post_headers = dict(Accept="application/json")
+                post_headers['Content-Type'] = 'application/json'
+
+                try:
+                    request_data = json.dumps(array_add_req)
```

Comment:
Probably should move the json.dumps() outside of this exception block.  Smaller exception blocks lead to more targetted exception handling and less surprises.